### PR TITLE
Show notifications for replied messages and handle oreo quick reply actions

### LINKED ISSUE #134: Watomatic doesn't work on Xiaomi devices
## ISSUE: Fails to work on Xiaomi device

Well, as said above, the app works fine on Xiaomi device with a major flaw.   
**It doesn't _Reply_ to any of the incoming messages.**    
  
May be, it's due to, different notification channel which is different from other OEMs, used by Xiaomi on its software **MIUI** as usual.   
  
### Device Specifications:
- **Android:** 10
- **MIUI:** 12.0.3.0
- **Model:** Redmi Note 7S

### LINKED ISSUE #189: Do not dismiss notification after auto reply
When Watomatic replies to a message, it is shown as "Read" in WhatsApp.
Would be very useful, when the message stays "Unread" or "New", so I know that there is a new message where Watomatic replied to.
This way I can check the message and see if I missed something.

### PR DESCRIPTION
**Pull request by spuday90**
closes #189

### CODE REVIEW

**Comment by spuday90** [comment]
Hi @adeekshith Please check on your device, on mine somehow getting 2 replies.

File: app/src/main/java/com/parishod/watomatic/NotificationService.java [hunk 1]
```diff
@@ -88,6 +94,42 @@ private void sendReply(StatusBarNotification sbn) {
...
+            actions.add(action);
+        }
+        List<RemoteInput> remoteInputs = new ArrayList<>();
+        PendingIntent pendingIntent = null;
+        for(NotificationCompat.Action act : actions) {
+            if(act != null && act.getRemoteInputs() != null
+                && remoteInputs.size() == 0 ) {
+                for(int x = 0; x < act.getRemoteInputs().length; x++) {
+                    RemoteInput remoteInput = act.getRemoteInputs()[x];
+                    if((remoteInput.getResultKey().toLowerCase().contains("reply")
```
Comment (adeekshith) [inline app/src/main/java/com/parishod/watomatic/NotificationService.java:111]: Using English word "reply" may not work for other languages

**Review by adeekshith — CHANGES_REQUESTED** [review]
Commented items and multiple replies needs to be fixed

File: app/src/main/java/com/parishod/watomatic/NotificationService.java (hunk 1 above)
Comment (spuday90) [reply app/src/main/java/com/parishod/watomatic/NotificationService.java]: Well actually i realised we dont need to check "reply" as remoteInput object only will for reply

**Comment by adeekshith** [comment]
@spuday90 
- Can we separate debug logs sharing code? It has external storage permissions so we might want to avoid that for now.
- What issues are we having with constrained layout?

**Comment by spuday90** [comment]
> @spuday90
> 
>     * Can we separate debug logs sharing code (cherry-pick)? It has external storage permissions so we might want to avoid that for now.
> 
>     * What issues are we having with constrained layout?

In the latest commits we have removed external storage permission and moved the logs write to internal storage
With constraint layout spacing was coming between main fragment content and dismiss notification settings, i think we can use biasing to fix that but felt linear layout is better and faster in that case so changed it

**Comment by adeekshith** [comment]
Also keeping the notification after reply only seems to be working for Whatsapp

**Comment by spuday90** [comment]
> Also keeping the notification after reply only seems to be working for Whatsapp

Oh!, have to debug which reply intent it is taking

**Comment by adeekshith** [comment]
> > Also keeping the notification after reply only seems to be working for Whatsapp
> 
> Oh!, have to debug which reply intent it is taking
> 

I think manually replying to facebook via notification also dismisses it so not really app's fault

**Comment by adeekshith** [comment]
@spuday90 conflicts

**Comment by adeekshith** [comment]
@spuday90 strings.xml and arrays.xml needs to be checked too. they replaced strings in main branch if not conflicting

**Review by adeekshith — CHANGES_REQUESTED** [review]
- Needs to be updated with latest "main" (merge/rebase)
- Resolve conflicts with main
- Fix strings.xml and arrays.xml if not already done

- Remove logging to file code

- Notification Setting
  - Remove "Oreo+ only" if it works on previous ones also
  - Move setting to "Settings activity"
  - Change switch label to "Dismiss replied notifications"
  - Remove help tip below the switch ("Turn On to dismiss...")

**Comment by adeekshith** [comment]
I see this in the latest branch with Watomatic attribution enabled

**Comment by adeekshith** [comment]
Also clicking on Notification button is behaving like "Share" button. Is this happening only on mine?
Tried deleting local branch and pulled latest.

https://user-images.githubusercontent.com/2568945/113635640-9e1c7e80-9636-11eb-8c54-d872ad623718.mp4

**Comment by adeekshith** [comment]
- Watomatic notification icon too small. Let me know if it needs anything done from from me end.
- Can we make it normal ("Alert" or similar) notification instead of Silent notification?
- WhatsApp original notification is still present. Should we dismiss that?

Screen Shot 2021-04-05 at 5 48 22 PM

**Comment by spuday90** [comment]
> Also clicking on Notification button is behaving like "Share" button. Is this happening only on mine?
> Tried deleting local branch and pulled latest.
> notification-btn-share.mp4

I completely removed share code in latest commit, so it should have crashed instead
If something was missed while removing that

**Comment by spuday90** [comment]
>     * Watomatic notification icon too small. Let me know if it needs anything done from from me end.
> 
>     * Can we make it normal ("Alert" or similar) notification instead of Silent notification?
> 
>     * WhatsApp original notification is still present. Should we dismiss that?
> 
> 
> Screen Shot 2021-04-05 at 5 48 22 PM

1. For notification icon I have used the foreground one for now, we should generate actual notification icon, or we can use app icon, App icon will look fine in status bar but when status bar pulled down then icon will look greyed out.
2.Yes we can change silent notification to standard notification
3.Yes we should dismiss whatsapp notification, as for other apps it is anyhow dismissing so itll be consistent

**Comment by adeekshith** [comment]
Have you seen this happening?
I am getting multiple replies on this branch

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -55,8 +55,17 @@
     <string name="watomatic_github_url" translatable="false">https://github.com/adeekshith/watomatic</string>
     <string name="watomatic_wato_message_url" translatable="false">https://www.reddit.com/r/watomatic/comments/lrkn7j/fellow_watos_share_your_creative_wato_message/</string>
 
+    <!-- Share debug logs -->
+    <string name="app_logs_file_name">WatomaticLogs.txt</string>
```
Comment (adeekshith) [inline app/src/main/res/values/strings.xml:59]: @spuday90 Mark all share debug logs as `translatable="false"`

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -55,8 +55,17 @@
...
+    <string name="share_email_body">Hi Team Watomatic,\nPlease find the attachment of the app logs.</string>
+    <string name="share_email_subject">Watomatic AppLogs</string>
+    <string name="share_email_id">watomatic@deekshith.in</string>
+    <string name="share_email_err_msg">Couldnot find any application to process this request</string>
+
     <!-- keys and app internal strings -->
     <string name="key_pref_app_language" translatable="false">pref_app_language</string>
+    <string name="pref_show_notification" translatable="false">pref_show_notification</string>
```
Comment (adeekshith) [inline app/src/main/res/values/strings.xml:67]: `pref_show_notification` -> `pref_show_notification_replied_msg` for value to future proof (but optional)

**Review by adeekshith — APPROVED** [review]

**Comment by adeekshith** [comment]
@spuday90 Approve this but I am still seeing these:
- Notifications are silent
- Clicking on notification is not opening any app

I remember you fixing it before but not sure. If that is not fixed, we can merge this PR and file an issue.

**Pull request merged**